src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(new Random(4000).nextLong());
+        } catch (InterruptedException e) {
+            System.out.println("Interrupted for " + userId);
```

Comment:
Interrupted exceptions should not be swallowed. If the thread's interrupted status has been set, it should be re-interrupted by calling `Thread.currentThread().interrupt()` to ensure that higher-level interrupt handling can take place.